<a href="https://colab.research.google.com/github/Adi11cr7/Atlas/blob/master/Atlas_3D_reconstruction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Steps to install conda 


This article has been referred : https://towardsdatascience.com/conda-google-colab-75f7c867a522


♒

In [ ]:
!which python # should return /usr/local/bin/python

/usr/local/bin/python


In [ ]:
!python --version

Python 3.7.12


In [ ]:
!echo $PYTHONPATH

/env/python


In [ ]:
%env PYTHONPATH=

env: PYTHONPATH=


In [ ]:
%%bash
MINICONDA_INSTALLER_SCRIPT=Miniconda3-4.5.4-Linux-x86_64.sh
MINICONDA_PREFIX=/usr/local
wget https://repo.continuum.io/miniconda/$MINICONDA_INSTALLER_SCRIPT
chmod +x $MINICONDA_INSTALLER_SCRIPT
./$MINICONDA_INSTALLER_SCRIPT -b -f -p $MINICONDA_PREFIX

PREFIX=/usr/local
installing: python-3.6.5-hc3d631a_2 ...
installing: ca-certificates-2018.03.07-0 ...
installing: conda-env-2.6.0-h36134e3_1 ...
installing: libgcc-ng-7.2.0-hdf63c60_3 ...
installing: libstdcxx-ng-7.2.0-hdf63c60_3 ...
installing: libffi-3.2.1-hd88cf55_4 ...
installing: ncurses-6.1-hf484d3e_0 ...
installing: openssl-1.0.2o-h20670df_0 ...
installing: tk-8.6.7-hc745277_3 ...
installing: xz-5.2.4-h14c3975_4 ...
installing: yaml-0.1.7-had09818_2 ...
installing: zlib-1.2.11-ha838bed_2 ...
installing: libedit-3.1.20170329-h6b74fdf_2 ...
installing: readline-7.0-ha6073c6_4 ...
installing: sqlite-3.23.1-he433501_0 ...
installing: asn1crypto-0.24.0-py36_0 ...
installing: certifi-2018.4.16-py36_0 ...
installing: chardet-3.0.4-py36h0f667ec_1 ...
installing: idna-2.6-py36h82fb2a8_1 ...
installing: pycosat-0.6.3-py36h0a5515d_0 ...
installing: pycparser-2.18-py36hf9f622e_1 ...
installing: pysocks-1.6.8-py36_0 ...
installing: ruamel_yaml-0.15.37-py36h14c3975_2 ...
installing: six-1.11

--2021-12-04 07:28:23--  https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh
Resolving repo.continuum.io (repo.continuum.io)... 104.18.201.79, 104.18.200.79, 2606:4700::6812:c94f, ...
Connecting to repo.continuum.io (repo.continuum.io)|104.18.201.79|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh [following]
--2021-12-04 07:28:23--  https://repo.anaconda.com/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh
Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.130.3, 104.16.131.3, 2606:4700::6810:8303, ...
Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.130.3|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 58468498 (56M) [application/x-sh]
Saving to: ‘Miniconda3-4.5.4-Linux-x86_64.sh’

     0K .......... .......... .......... .......... ..........  0% 43.1M 1s
    50K .......... .......... .......... .......... ..........  0%

In [ ]:

!which conda # should return /usr/local/bin/conda

/usr/local/bin/conda


In [ ]:


!conda --version # should return 4.5.4

conda 4.5.4


In [ ]:
!python --version

Python 3.6.5 :: Anaconda, Inc.


In [ ]:
!pip install tqdm

    100% |████████████████████████████████| 81kB 4.9MB/s 


In [ ]:
%%bash
conda install --channel defaults conda python=3.6 --yes
conda update --channel defaults --all --yes

Solving environment: ...working... done

## Package Plan ##

  environment location: /usr/local

  added / updated specs: 
    - conda
    - python=3.6


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    python-3.6.13              |       h12debd9_1        32.5 MB
    tqdm-4.62.3                |     pyhd3eb1b0_1          79 KB
    pycosat-0.6.3              |   py36h27cfd23_0         107 KB
    charset-normalizer-2.0.4   |     pyhd3eb1b0_0          33 KB
    tk-8.6.11                  |       h1ccaba5_0         3.2 MB
    ca-certificates-2021.10.26 |       h06a4308_2         121 KB
    pycparser-2.21             |     pyhd3eb1b0_0          94 KB
    zlib-1.2.11                |       h7b6447c_3         120 KB
    conda-package-handling-1.7.3|   py36h27cfd23_1         946 KB
    brotlipy-0.7.0             |py36h27cfd23_1003         349 KB
    six-1.16.0                 |     pyhd3eb1b0_0  

python-3.6.13        | 32.5 MB | ########## | 100% 
tqdm-4.62.3          |   79 KB | ########## | 100% 
pycosat-0.6.3        |  107 KB | ########## | 100% 
charset-normalizer-2 |   33 KB | ########## | 100% 
tk-8.6.11            |  3.2 MB | ########## | 100% 
ca-certificates-2021 |  121 KB | ########## | 100% 
pycparser-2.21       |   94 KB | ########## | 100% 
zlib-1.2.11          |  120 KB | ########## | 100% 
conda-package-handli |  946 KB | ########## | 100% 
brotlipy-0.7.0       |  349 KB | ########## | 100% 
six-1.16.0           |   18 KB | ########## | 100% 
yaml-0.2.5           |   87 KB | ########## | 100% 
idna-3.3             |   55 KB | ########## | 100% 
urllib3-1.26.7       |  104 KB | ########## | 100% 
requests-2.26.0      |   59 KB | ########## | 100% 
ld_impl_linux-64-2.3 |  637 KB | ########## | 100% 
pip-21.2.2           |  2.1 MB | ########## | 100% 
libgcc-ng-9.1.0      |  8.1 MB | ########## | 100% 
wheel-0.37.0         |   31 KB | ########## | 100% 
setuptools-5

In [ ]:
!conda --version # now returns 4.8.3

conda 4.10.3


In [ ]:
import sys
_ = (sys.path
        .append("/usr/local/lib/python3.6/site-packages"))

In [ ]:
!conda install -y pytorch=1.5.0 torchvision=0.6.0 cudatoolkit=10.2 -c pytorch --yes

Solving environment: | / - \ | / - \ | / - \ | / - \ failed with initial frozen solve. Retrying with flexible solve.
Solving environment: / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / failed with repodata from current_repodata.json, will retry with next repodata source.
Solving environment: / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - cudatoolkit=10.2
    - pytorch=1.5.0
    - torchvision=0.6.0


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    blas-1.0                   |              mkl           6 KB
    cudatoolkit-10.2.89        |     

In [ ]:
!conda install opencv --yes

Solving environment: \ | / - \ | / - \ | / done

## Package Plan ##

  environment location: /usr/local

  added / updated specs:
    - opencv


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    bzip2-1.0.8                |       h7b6447c_0          78 KB
    cairo-1.16.0               |       hf32fb01_1         1.0 MB
    ffmpeg-4.0                 |       hcdf2ecd_0        53.3 MB
    fontconfig-2.13.1          |       h6c09931_0         250 KB
    freeglut-3.0.0             |       hf484d3e_5         176 KB
    glib-2.69.1                |       h5202010_0         1.7 MB
    graphite2-1.3.14           |       h23475e2_0          99 KB
    harfbuzz-1.8.8             |       hffaf4a1_0         507 KB
    hdf5-1.10.2                |       hba1933b_1         3.8 MB
    icu-58.2                   |       he6710b0_3        10.5 MB
    jasper-2.0.14              |       

In [ ]:
!pip install \
  open3d>=0.10.0.0 \
  trimesh>=3.7.6 \
  pyquaternion>=0.9.5 \
  pytorch-lightning>=0.8.5 \
  pyrender>=0.1.43

In [ ]:
!pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu102/torch1.5/index.html

Looking in links: https://dl.fbaipublicfiles.com/detectron2/wheels/cu102/torch1.5/index.html
     |████████████████████████████████| 6.9 MB 61.5 MB/s 
     |████████████████████████████████| 106 kB 6.2 MB/s 
     |████████████████████████████████| 49 kB 5.5 MB/s 
  Using cached Cython-0.29.24-cp36-cp36m-manylinux1_x86_64.whl (2.0 MB)
  Created wheel for fvcore: filename=fvcore-0.1.5.post20211023-py3-none-any.whl size=60947 sha256=9763a1341692c9ea8e8a6f2caf34fc588dea6b7b494859135bf00549b2a12a30
  Stored in directory: /root/.cache/pip/wheels/bf/b8/6b/54671f269454abdaf50414f567d35ed617ee7289189ec8013c
  Created wheel for pycocotools: filename=pycocotools-2.0.3-cp36-cp36m-linux_x86_64.whl size=274304 sha256=3963cac9c7e9b679f83a6a077c6eacf0627c055b4bce3b4e9be1d5478b1c5213
  Stored in directory: /root/.cache/pip/wheels/af/2e/64/1c2780138475c28ca8649d218b6f3ddb5a7dadb41f894002bd
  Created wheel for termcolor: filename=termcolor-1.1.0-py3-none-any.whl size=4847 sha256=5fbb0afe834ffc662c7346046

In [ ]:
import os
from google.colab import files

%cd

# Clone the repo
!git clone https://github.com/magicleap/Atlas.git
%cd Atlas

# Download pretrained weights
if not os.path.exists('results'):
  !wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=12P29x6revvNWREdZ01ufJwMFPl-FEI_V' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=12P29x6revvNWREdZ01ufJwMFPl-FEI_V" -O results.zip && rm -rf /tmp/cookies.txt
  !unzip results.zip

# Download sample data
if not os.path.exists('data'):
  !wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=13-D7QNVZjj864E768zJ7IWDDqY6agUES' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=13-D7QNVZjj864E768zJ7IWDDqY6agUES" -O data.zip && rm -rf /tmp/cookies.txt
  !unzip data.zip

# verify everything was downloaded
!ls
!ls results/release/semseg
!ls data/sample/sample1



/root
fatal: destination path 'Atlas' already exists and is not an empty directory.
/root/Atlas
--2021-12-04 07:44:17--  https://docs.google.com/uc?export=download&confirm=hR2M&id=13-D7QNVZjj864E768zJ7IWDDqY6agUES
Resolving docs.google.com (docs.google.com)... 64.233.189.102, 64.233.189.113, 64.233.189.100, ...
Connecting to docs.google.com (docs.google.com)|64.233.189.102|:443... connected.
HTTP request sent, awaiting response... 302 Moved Temporarily
Location: https://doc-0k-4k-docs.googleusercontent.com/docs/securesc/94qb4eat3ss2g86lc4tsdk3v9mq4mkr7/8141v9tu8eplou2dfpp7rh9m4mbdk2n9/1638603825000/06775661448174390731/05884305935771865020Z/13-D7QNVZjj864E768zJ7IWDDqY6agUES?e=download [following]
--2021-12-04 07:44:18--  https://doc-0k-4k-docs.googleusercontent.com/docs/securesc/94qb4eat3ss2g86lc4tsdk3v9mq4mkr7/8141v9tu8eplou2dfpp7rh9m4mbdk2n9/1638603825000/06775661448174390731/05884305935771865020Z/13-D7QNVZjj864E768zJ7IWDDqY6agUES?e=download
Resolving doc-0k-4k-docs.googleusercontent

In [ ]:
!rm -rf data/sample/sample1/color

In [ ]:
!rm -rf data/sample/sample1/pose

In [ ]:
!unzip  data/sample/sample1/color.zip -d data/sample/sample1

Archive:  data/sample/sample1/color.zip
   creating: data/sample/sample1/content/color/
  inflating: data/sample/sample1/content/color/00000025.jpg  
  inflating: data/sample/sample1/content/color/00000019.jpg  
  inflating: data/sample/sample1/content/color/00000004.jpg  
  inflating: data/sample/sample1/content/color/00000021.jpg  
  inflating: data/sample/sample1/content/color/00000007.jpg  
  inflating: data/sample/sample1/content/color/00000011.jpg  
  inflating: data/sample/sample1/content/color/00000017.jpg  
  inflating: data/sample/sample1/content/color/00000009.jpg  
  inflating: data/sample/sample1/content/color/00000012.jpg  
  inflating: data/sample/sample1/content/color/00000002.jpg  
  inflating: data/sample/sample1/content/color/00000014.jpg  
  inflating: data/sample/sample1/content/color/00000001.jpg  
  inflating: data/sample/sample1/content/color/00000013.jpg  
  inflating: data/sample/sample1/content/color/00000006.jpg  
  inflating: data/sample/sample1/content/col

In [ ]:
!pip install scikit-image

In [ ]:
# prepare data
!python prepare_data.py --path data --path_meta data --dataset sample

preparing sample1
preparing sample1 frame 0/250
preparing sample1 frame 25/250
preparing sample1 frame 50/250
preparing sample1 frame 75/250
preparing sample1 frame 100/250
preparing sample1 frame 125/250
preparing sample1 frame 150/250
preparing sample1 frame 175/250
preparing sample1 frame 200/250
preparing sample1 frame 225/250


In [ ]:
def standardize_name(filename):
    f, s = filename.split(".")
    f = f.zfill(8)
    return f+"."+s

In [ ]:
import os
path = "data/sample/sample1/pose/"
for filename in os.listdir("data/sample/sample1/pose"):
      os.rename(path+filename, path+standardize_name(filename))
    
   

In [ ]:
import os
path = "data/sample/sample1/color/"
for filename in os.listdir("data/sample/sample1/color"):
      os.rename(path+filename, path+standardize_name(filename))

In [ ]:
!pip uninstall torchvision        -- yes
!pip install torchvision==0.6.0
!pip install torch==1.5.0 pytorch-lightning>=0.8.5

Found existing installation: torchvision 0.6.0a0+82fd1c8
Uninstalling torchvision-0.6.0a0+82fd1c8:
  Would remove:
    /usr/local/lib/python3.6/site-packages/torchvision
    /usr/local/lib/python3.6/site-packages/torchvision-0.6.0a0+82fd1c8-py3.6.egg-info
Proceed (Y/n)? y
  Successfully uninstalled torchvision-0.6.0a0+82fd1c8
     |████████████████████████████████| 6.6 MB 3.7 MB/s 
     |████████████████████████████████| 752.0 MB 3.4 kB/s 
  Attempting uninstall: torch
    Found existing installation: torch 1.10.0
    Uninstalling torch-1.10.0:
      Successfully uninstalled torch-1.10.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pytorch-lightning 1.5.4 requires torch>=1.7.*, but you have torch 1.5.0 which is incompatible.


# Important Change to make

Before running inference, navigate to the file /root/Atlas/atlas/model.py

**On line 92**, replace the line 

```
self.hparams = hparams
```
with 

 

```
 for key in hparams.keys():
    self.hparams[key]=hparams[key]
```







On line **107** in **inference.py** replace it with the line





 ```mesh_pred.export(os.path.join(save_path, '%s.obj'%scene))```



In [ ]:
!pip install protobuf==3.5.1

     |████████████████████████████████| 6.4 MB 4.3 MB/s 
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.19.1
    Uninstalling protobuf-3.19.1:
      Successfully uninstalled protobuf-3.19.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tensorboard 2.7.0 requires protobuf>=3.6.0, but you have protobuf 3.5.1 which is incompatible.


In [ ]:
#Date 04-12-2021
#TypeError: __new__() got an unexpected keyword argument 'serialized_options'
#Try updating protobuf to 3.6
!pip install protobuf==3.6

     |████████████████████████████████| 7.1 MB 3.5 MB/s 
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.19.1
    Uninstalling protobuf-3.19.1:
      Successfully uninstalled protobuf-3.19.1


In [ ]:
# run inference
!python inference.py --model results/release/semseg/final.ckpt --scenes data/sample/sample1/info.json --voxel_dim 208 208 80

0 1 sample sample1 0 250
0 1 sample sample1 25 250
0 1 sample sample1 50 250
0 1 sample sample1 75 250
0 1 sample sample1 100 250
0 1 sample sample1 125 250
0 1 sample sample1 150 250
0 1 sample sample1 175 250
0 1 sample sample1 200 250
0 1 sample sample1 225 250


In [ ]:
!conda install x264=='1!152.20180717' ffmpeg=4.0.2 -c conda-forge --yes

Solving environment: | failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / - \ | / 

In [ ]:
# save input images as video
#!ffmpeg -r 10 -f image2 -s 1920x1080 -i data/sample/sample1/color/%08d.jpg -vcodec libx264 -crf 25  -pix_fmt yuv420p results/release/semseg/test_final/sample1.mp4
!ffmpeg -r 10 -f image2 -s 640x480 -i data/sample/sample1/color/%08d.jpg -vcodec libx264 -crf 25  -pix_fmt yuv420p results/release/semseg/test_final/sample1.mp4

ffmpeg version 4.0.2 Copyright (c) 2000-2018 the FFmpeg developers
  built with gcc 4.8.2 (GCC) 20140120 (Red Hat 4.8.2-15)
  configuration: --prefix=/usr/local --disable-doc --disable-openssl --enable-shared --enable-static --extra-cflags='-Wall -g -m64 -pipe -O3 -march=x86-64 -fPIC' --extra-cxxflags='-Wall -g -m64 -pipe -O3 -march=x86-64 -fPIC' --extra-libs='-lpthread -lm -lz' --enable-zlib --enable-pic --enable-pthreads --enable-gpl --enable-version3 --enable-hardcoded-tables --enable-avresample --enable-libfreetype --enable-gnutls --enable-libx264 --enable-libopenh264
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
Input #0, image2, from 'data/sample/sample1/color/%0

**For obj mesh refer this link** : https://github.com/magicleap/Atlas/issues/17

In [ ]:
files.download('results/release/semseg/test_final/sample1.obj')
files.download('results/release/semseg/test_final/sample1.mp4')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>